labs/aws/csharp/lab-04/README.md

```diff
@@ -0,0 +1,363 @@
+# Deploying Containers to a Kubernetes Cluster
+
+In this lab, you will deploy a containerized application to a Kubernetes cluster.
+
+> This lab assumes you have a project set up. If you don't yet, please [complete this lab first](../01-iac/01-creating-a-new-project.md).
+
+## Step 1 &mdash; Creating a Kubernetes Cluster
+
+> If you do not have an EKS cluster, you can create one by using the code [here](./code/step1.cs).
+
+If you have created your EKS Cluster using Pulumi, then you can export your KubeConfig as follows:
+
+```bash
+pulumi stack output kubeconfig > ~/iac-workshop/kubeconfig.json
+```
+
+Point the `KUBECONFIG` environment variable at your cluster configuration file:
+
+```bash
+export KUBECONFIG=~/iac-workshop/kubeconfig
+```
+
+To test out connectivity, run `kubectl cluster-info`. You should see information similar to this:
+
+```
+Kubernetes master is running at https://abcxyz123.gr7.eu-central-1.eks.amazonaws.com
+CoreDNS is running at https://abcxyz123.gr7.eu-central-1.eks.amazonaws.com/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+```
+
+## Step 2 &mdash; Install the Kubernetes Package
+
+From your project's root directory, install the Kubernetes package:
+
+```
+dotnet add package Pulumi.Kubernetes --version 1.6.0-preview
+```
+
+Next, add these imports to your `MyStack.cs` file:
+
+```csharp
+using CoreV1 = Pulumi.Kubernetes.Core.V1;
+using AppsV1 = Pulumi.Kubernetes.Apps.V1;
+using Pulumi.Kubernetes.Types.Inputs.Apps.V1;
+using Pulumi.Kubernetes.Types.Inputs.Core.V1;
+using Pulumi.Kubernetes.Types.Inputs.Meta.V1;
+```
+
+We need to declare a new Kubernetes Provider based on the KubeConfig created in step1. To do this, add this to your `MyStack.cs` file
+
+```csharp
+var k8sProvider = new K8s.Provider("k8s-provider", new K8s.ProviderArgs
+{
+    KubeConfig = this.Kubeconfig
+}, new CustomResourceOptions
+{
+    DependsOn = {nodeGroup},
```

Comment:
nit: kubeconfig should dependOn the cluster as it is a property of it, though i suspect this was done to ensure nodes have been created first?

Reply:
exactly - it's only to ensure that the nodes are created - otherwise the rest of the application tries to run and fails creating the NS because the cluster isn't ready